protobuf/model_config.proto

```diff
@@ -239,6 +239,15 @@ message ModelInstanceGroup
   //@@     not be added to the scheduler. Default value is false.
   //@@
   bool passive = 7;
+
+  //@@  .. cpp:var:: int32 numa_id
+  //@@
+  //@@     The id used to associate with NUMA related features for the instance
+  //@@     group. For KIND_GPU, this field will be ignored and the value will be
```

Comment:
Instead of ignoring, make it an error to explicitly specify numa_id for KIND_GPU.

Reply:
After some thinking, we should be able to have a "device" field, which refers to NUMA specific ID for CPU instance and GPU device id for GPU instance, instead of "numa_id". This was discussed before, but actually the backward compatibility concern may not apply here:

There are actually two notions: "memory type and id" that describes the memory property and "instance kind and id" describes the instance property. I mixed them together before because they are the same in GPU case. And the backend compatibility concern applies to the memory description where one may implements a different frontend / backend that relies on the "memory type and id" even in CPU case, so we shouldn't add more meanings to it.

But for the "instance kind and id", allowing CPU instance to have id different from 0 shouldn't break any existing code. Because the return value of `TRITONBACKEND_ModelInstanceDeviceId()` is not in control of the user and it is always 0 for CPU instance, it is not likely that there is logic that depends on a constant value.

That being said, we still need to extend the memory related APIs so it may return requested memory from location closer to specified device. The difference is that the device will be specified by "device kind / device id" pair but not "device kind / numa id" pair, which is more general.